fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -954,6 +956,26 @@ public Void visitLikePredicate(LikePredicate node, Scope scope) {
             return null;
         }
 
+        @Override
+        public Void visitMatchExpr(MatchExpr node, Scope scope) {
+            Type type1 = node.getChild(0).getType();
+            Type type2 = node.getChild(1).getType();
+
+            if (!type1.isStringType() && !type1.isNull()) {
+                throw new SemanticException("left operand of MATCH must be of type STRING with NOT NULL");
+            }
+
+            if (!(node.getChild(0) instanceof SlotRef)) {
+                throw new SemanticException("left operand of MATCH must be column ref");
+            }
+
+            if (!type2.isStringType() && !type2.isNull()) {
```

Comment:
Is the sceond parameter should be a constant string value?

Reply:
I think string type is fine, if the second parameter is a function with return type as string, it is also avaliable.